### ISSUE TITLE
When starting a server and it failing to connect to an IP/Port, and it asking new information, and then it connecting fine, the server just closes.

### ISSUE THREAD

**Issue by MichaelAgarkov** [MichaelAgarkov/MicChatRevived#2]
### When starting a server and it failing to connect to an IP/Port, and it asking new information, and then it connecting fine, the server just closes.
This is not a classic MicChat issue, because the classic MicChat server didn't ask for new information after failing to connect, and it just closed. I might have forgotten to remove something from the code for when it fails to connect, which makes it do it. I thought I removed every `exit;` from there, but something else might do that. This is indeed a bug and has to be fixed.

**Comment by MichaelAgarkov** [comment]
It most likely has to do something with **this**:
`if(ErrorOccured = false) then while true do
        if listener.Pending then
        begin
          i += 1;
          n := i;
          client[i] := listener.AcceptTcpClient();
          thr[i] := new Thread(chat);
          thr[i].Start(i);
        end;
    end until Connected;
end.`

**Comment by MichaelAgarkov** [comment]
This most likely has to do something with **this**:

- if(ErrorOccured = false) then while true do
-         if listener.Pending then
-         begin
-           i += 1;
-           n := i;
-           client[i] := listener.AcceptTcpClient();
-           thr[i] := new Thread(chat);
-           thr[i].Start(i);
-         end;
-     end until Connected;
- end.

**Comment by MichaelAgarkov** [comment]
This most likely has to do something with **this**:

```if(ErrorOccured = false) then while true do
        if listener.Pending then
        begin
          i += 1;
          n := i;
          client[i] := listener.AcceptTcpClient();
          thr[i] := new Thread(chat);
          thr[i].Start(i);
        end;
    end until Connected;
end.```

**Comment by MichaelAgarkov** [comment]
This most likely has to do something with **this**:

```
if(ErrorOccured = false) then while true do
        if listener.Pending then
        begin
          i += 1;
          n := i;
          client[i] := listener.AcceptTcpClient();
          thr[i] := new Thread(chat);
          thr[i].Start(i);
        end;
    end until Connected;
end.
```

**Comment by MichaelAgarkov** [comment]
Fixed in update 1.3
